resources/splunk_install.rb

```diff
@@ -0,0 +1,219 @@
+    state['packages'] ||= {}
+
+    raise "Installation resource with name #{name} already exists!" if state['packages'][name]
+    state['current_package'] = state['packages'][name] = {
+      package: package,
+      version: version,
+      build: build,
+      x64: x64_support
+    }
+
+    converge_if_changed :version, :build do
+      remote_file package_path.to_s do
+        source package_url
+        show_progress true if defined? show_progress # Chef 12.9 feature
+        notifies :delete, "remote_file[#{package_path}]", :delayed
+      end
+    end
+
+    user service_user do
+      password service_password if service_password
+      system true
+      action :create
+    end
+  end
+
+  action :uninstall do
+    directory install_dir do
+      recursive true
+      action :delete
+    end
+  end
+end
+
+###################################
+### Platform Specific Providers ###
+###################################
+
+class LinuxInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, os: 'linux'
+
+  def kernel_string
+    x64_support ? 'Linux-x86_64.tgz' : 'Linux-i686.tgz'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      tar_extract package_path do
+        action :extract_local
+        target_dir install_dir
+      end
+    end
+  end
+end
+
+class RedhatInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, platform_family: 'rhel'
+
+  def kernel_string
+    x64_support ? 'linux-2.6-x86_64.rpm' : 'i386.rpm'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      rpm_package package_name do
+        source package_path.to_s
+        action :install
+      end
+    end
+  end
+
+  action :uninstall do
+    rpm_package package_name do
+      action :remove
+    end
+  end
+end
+
+class DebianInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, platform_family: 'debian'
+
+  def kernel_string
+    x64_support ? 'linux-2.6-amd64.deb' : 'linux-2.6-intel.deb'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      dpkg_package package_name do
+        source package_path.to_s
+        action :install
+      end
+    end
+  end
+
+  action :uninstall do
+    dpkg_package package_name do
+      action :purge
+    end
+  end
+end
+
+class WindowsInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, os: 'windows'
+
+  def package_name
+    case package
+    when :splunk then 'Splunk Enterprise'
+    when :universal_forwarder then 'UniversalForwarder'
+    end
+  end
+
+  # kernel_string = x64_support ? 'windows-64' : 'windows-32'
+  # extension = 'zip'
+  def kernel_string
+    x64_support ? 'x64-release.msi' : 'x86-release.msi'
+  end
+
+  action :install do
+    raise 'Windows installation will fail unless license is accepted.' unless accept_license
```

Comment:
Do we even want to provide the option to not programmatically accept the license then?